cmake/modules/RootBuildOptions.cmake

```diff
@@ -127,6 +127,7 @@ ROOT_BUILD_OPTION(jemalloc OFF "Using the jemalloc allocator")
 ROOT_BUILD_OPTION(krb5 OFF "Kerberos5 support, requires Kerberos libs")
 ROOT_BUILD_OPTION(ldap OFF "LDAP support, requires (Open)LDAP libs")
 ROOT_BUILD_OPTION(libcxx OFF "Build using libc++, requires cxx11 option (MacOS X only, for the time being)")
+ROOT_BUILD_OPTION(lz4 ON "LZ4 compression algorithm as a default, requires liblz4")
```

Comment:
The option semantic/documentation and its name do not match well.  Usually ItemName=ON means to generate the library for/using this items (see for example the two option above).  Should we use something more lilke: ```lz4_as_default``` or ```compression_default=lz4```?

Reply:
Thank you for advise, I like both options, maybe second is more flexible, what you think?

Reply:
I selected first case, but I think  maybe in a case of support more compression algorithms and also changes proposed in https://github.com/root-project/root/pull/1637, I can do a second iteration together with PR#1637

Reply:
I also think that the 2nd (compression_default=lz4) is better and flexible and would lean for this one.  (If we keep the 1st one we *do need* to change the spelling as building lz4 is not optional)

Reply:
@pcanal I updated and tested locally according your recommendations..